### PR TITLE
[FRON-1464]: Added customer name to the metadata.

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Add customer name to the metadata

Jira Ticket: https://omise.atlassian.net/browse/FRON-1464

#### 2. Description of change

Customer name is fetched from the `$order` object. A new function `getOrderMetadata` which is created in the `class-omise-payment.php` that abstracts the code to generate the metadata to be sent to the OMISE server. It takes `$order_id` and `$order` as the parameters.

Even though we could obtain the order ID from the `$order` obejct, we passed the `$order_id` to the `getOrderMetada` method to avoid checking for backward compatibility.

```
/** override order_id as a reference for webhook handlers **/
/** backward compatible with WooCommerce v2.x series **/
'order_id' => version_compare( WC()->version, '3.0.0', '>=' ) ? $order->get_id() : $order->id
```

#### 3. Quality assurance

Add item to the cart and checkout using following payment methods
- Alipay
- Alipay Plus
- Bill payment Tesco
- Credit Card
- FPX
- Installment
- Internet banking
- Konbini
- Mobile banking
- Rabbit linepay
- True Money

Tested with following payment methods
Screen Shot 2565-05-09 at 16 56 08

On the dashboard, go to any one charge and you should see the customer name in the metadata.

Screen Shot 2565-05-09 at 16 45 44

**🔧 Environments:**

- WooCommerce: v6.4.1
- WordPress: v5.9.3
- PHP version: 7.4.28
- Omise plugin version: Omise-WooCommerce 4.19.2

### CODE REVIEW

**Comment by FhanOmise** [comment]
Need to update this on `includes/gateway/class-omise-payment-ocbc-pao.php` from `https://github.com/omise/omise-woocommerce/pull/265` as well. 🤔

**Review by FhanOmise — APPROVED** [review]

**Comment by tanawin-opn** [comment]
I've tested with my website but even I'm filled all the billing address

image

the metadata here still empty value

image

**Comment by aashishgurung** [comment]
> 

Thanks for reporting it. It should have worked. I will ping you via slack to know more about it.

**Comment by kan98** [comment]
https://github.com/omise/omise-woocommerce/pull/270 if you release after me, can you incorporate your change on GrabPay (new payment method) too?

**Comment by aashishgurung** [comment]
> #270 if you release after me, can you incorporate your change on GrabPay (new payment method) too?

Sure

**Comment by tanawin-opn** [comment]
retest again with billing address fixed, now it works

image

**Review by kan98 — APPROVED** [review]

File: includes/gateway/class-omise-payment-installment.php [hunk 1]
```diff
@@ -104,7 +101,7 @@ public function charge( $order_id, $order ) {
 				'installment_terms' => sanitize_text_field( $installment_terms )
 			),
 			'return_uri'        => $return_uri,
-			'metadata'          => $metadata
+			'metadata'    => $this->getOrderMetadata($order_id, $order)
```
Comment (hpariwat) [inline includes/gateway/class-omise-payment-installment.php:104]: The indentation seems to be a bit off.

Comment (aashishgurung) [reply includes/gateway/class-omise-payment-installment.php]: Fixed. Thanks.

**Review by new4762 — APPROVED** [review]

**Review by hpariwat — APPROVED** [review]

**Pull request closed without merging**